powerhome/message_bus
# middleware tracing

- These methods can all be private
- Three ways to close a client socket are unnecessary
- Documents the MessageBus::Client interface
- ConnectionManager#stats is unused
- Documents ConnectionManager
- Much of MessageBus::Rack::Middleware can be private
- Documents MessageBus::Rack::Middleware
- Much of the diagnostics implementation can be private
- Documents diagnostics interface
- Puts existing documentation in the right place
- These methods can be private
- Documents MessageBus::Implementation interface
- Use Client#close rather than Client#cancel
- don't start message bus in rake tasks
- fix crashes when destroying message bus
- Makes this an opt-in change
- Formatting
- Test without mocks
- Allow destroying a bus that never had a backend
- Raises rubocop limit on Metrics/ModuleLength
- Simple usage documentation
- Keep a running changelog of changes on master since the last release
- Mention changelog entries in contributing docs
- FIX: stop leaking redis keys used for tracking per channel id
- release 2.2.0.pre
- FIX: private method should not be invoked on self
- Re-generate rubocop todo list
- Inherit rubocop off discourse
- Generate backend-specific spec tasks automatically
- Benchmark publication performance
- Run variations on publish benchmark
- Notes on performance tests in the README
- Make it easy to manually test the diagnostics UI
- Re-writes diagnostics UI using React
- More diagnostics documentation
- FEATURE: Ruby MessageBus HTTP client.
- Release 2.2.0.pre.2.
- Specify units for Ruby HTTP Client configs.
- FIX: Ruby HTTP Client not backing off polling on error.
- Release 2.2.0.
- update README
- on_middleware_error only for internal errors
- Removes trailing whitespace
- Tests for handling of downstream errors
- bump version for release
- Print the full stack when message bus callback fails.
- Fix to make the `is_readonly?` method compatible with the redis gem both pre and post v4.0 when the `client` attribute was removed - closes #208
- Run prettier on file so it confirms to a consistent style
- FEATURE: support status 429 from server and Retry-After header
- FIX: Protect Redis config from being manipulated
- Update changelog and bump version
- FIX: remove deperecated Minitest global assertions
- FIX: Magic comment to satisfy rubocop
- FIX: Code adjustment to unfreeze strings
- Revert "FIX: remove deperecated Minitest global assertions"
- FIX: Use minitest-global_expectations
- README.md: Doc adding extra response headers
- README: Link to generated docs
- FEATURE: add optional shouldLongPollCallback
- bump version
- DEV: Add missing Rubocop gem dependencies due to changes in discourse config
- DEV: Fix broken rubocop rules
- Run tests against Ruby 2.6 on CI.
- Support for Ruby 2.3 has ended
- FIX: Don't publish message to intersection of `user_ids` and `group_ids`
- DEV: cut a new release
- FEATURE: `MessageBus#register_client_message_filter`
- DEV: bump version
- FIX: compatibility with Rails 6.0.3
- README: Add missing date to CHANGELOG
- FEATURE: `MessageBus.base_route`
- Use start_with to check route
- FIX: Remove regex, use substring
- Memoize the bus routes
- Clean up Javascript (#225)
- FEATURE: Support UMD and NPM (#226)
- Version bump for mew UMD js
- Add npm publish to release task
- Update Changelog
- FIX: Avoid leaking Redis connections. (#229)
- Release 3.3.1.
- FIX: Disconnect Redis conn when rescuing errors in global subscribe.
- Bump version to 3.3.2.
- Bump version to 3.3.2.
- Revert release of 3.3.1 and 3.3.2.
- Follow up to c0ad2809231260f24007a39a11ce73fecafddb8a.
- Release 3.3.1.
- PERF: optimize performance of message generation
- remove confusing comment
- FEATURE: raise when attempting to publish to invalid targets
- BREAKING Make JS client throw if lastId not number
- Prettify error backtrace when logging.
- Log when DistributedCache encounters an error when publishing.
- Release 3.3.2.
- Add Ruby 2.7 to Travis and remove Ruby 2.4.
- Fix failing specs due when stopping `MessageBus::TimerThread`.
- FIX: `queue_in_memory` option not being passed to the backends.
- FIX: `MessageBus::DistributedCache#publish` should raise on error.
- Add missing changelog for e91d41483a89dbcd3a6ecc3aeef47d353b5c6681.
- FEATURE: Add `publish_queue_in_memory` option to distributed cache manager.
- Release 3.3.3.
- FIX: Remove trailing comma incorrectly added in ec60d8865.
- Release 3.3.4.
- DEV: removes dead code
- FEATURE: Drop support for ruby 2.3 (#239)
- DEV: Switch from Travis to Actions (#240)
- DEV: Fix Rubocop runs on Github Actions (#241)
- FIX: force a poll more consistently when visibility changes
- Remove live demo, it has been removed
- Allow tracing the performance of middleware
